### ISSUE TITLE
WM requirements?

### ISSUE THREAD

**Issue by p4vook** [vimpostor/blobdrop#4]
What WM does blobdrop require?

Can I use it with KDE Plasma 5, specifically?

It would be nice to have this info in the readme.

**Comment by p4vook** [comment]
NVM about me, I thought blobdrop in-place feature would work on Wayland, but it probably won't. 

Documentation would still be nice though...

**Comment by vimpostor** [comment]
> What WM does blobdrop require?

It will work with any WM by default (X11 or Wayland). Full support for **all frontend options** is available "only" on all EWMH-compliant WMs.

> in-place feature won't work on Wayland

You are correct, Wayland [does not support starting drag operations programmatically without an implicit grab](https://wayland.freedesktop.org/docs/html/apa.html#protocol-spec-wl_data_device), this is why `--frontend immediate` won't work there.
Feel free to ask the Wayland devs to implement yet another workaround protocol for this usecase, but personally I wouldn't get my hopes up seeing how even [much more reasonable protocol requests](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/216) are being bikeshed for years in the review stage.

> Documentation would still be nice though...

This is already documented in the [man page](https://github.com/vimpostor/blobdrop/blob/9c5d2cc5ea3f98d4344bd5d76d77dc361dbfd484/doc/man/man1/blobdrop.1#L85) and [in the CLI](https://github.com/vimpostor/blobdrop/blob/9c5d2cc5ea3f98d4344bd5d76d77dc361dbfd484/src/getopts.cpp#L68-L69) when you try to force the immediate drag frontend, in other words you already get a very clear warning when you try to use a frontend on an unsupported platform.

If you really want to use the immediate drag frontend (and in case you still aren't disillusioned by the mess going on in Wayland and just switch back to X11), you can try running blobdrop over XWayland, but you will likely only be able to drop into other XWayland windows (urging the question, how can a compatibility layer still be so broken after so many years?).

And because the default frontend is always chosen to be one that is guaranteed to work on the user's system (on Wayland you get the popup window, which works just fine), I see no reason to put a disclaimer in the README. Otherwise I would spend all day documenting Wayland caveats, given how fundamentally broken Wayland's design is:
E.g. the [--cursor option](https://github.com/vimpostor/blobdrop/blob/9c5d2cc5ea3f98d4344bd5d76d77dc361dbfd484/src/backend.cpp#L148) won't work on Wayland too (applications are not even allowed to know their own position lol), and the automatic [temporary hiding of the parent terminal window](https://github.com/vimpostor/blobdrop/blob/9c5d2cc5ea3f98d4344bd5d76d77dc361dbfd484/src/backend.cpp#L28) is impossible to implement on Wayland for "security" reasons. Wayland's extreme "policy over mechanism" stance is incompatible with niche workflow tools like this one and many others.
I have already made sure that on Wayland the default frontend is changed to one that works, I don't think there is anything else I can (and quite frankly should) do.

**Comment by p4vook** [comment]
Yes, I've read your quite descriptive commit disabling immediate frontend on Wayland, and no, I'm not too eager to propose anything to Wayland.

I was basically asking about adding information on WM and Wayland support in README, as that's IMO the most convenient place for such information (because man pages are usually read only after installation to figure out the exact details of usage). 

By the way, I want to say that such descriptive and motivated comments are a rarity in open source, and it would be much easier to navigate the space, if everyone wrote them. Nice job!

**Comment by vimpostor** [comment]
> I was basically asking about adding information on WM and Wayland support in README

Well it's only one frontend that is completely borked on Wayland, most others still work fine so it doesn't make sense to put a big warning in the readme. And people who use an experimental display server like Wayland, should be aware themself that some workflows will be broken.

Nevertheless I have added a short comment now to the frontends list.

> such descriptive and motivated comments are a rarity in open source

Thanks, but it's less a case of "being motivated" and more a case of "utterly  bewildered by how broken this technology is".
